internal/statemachine/helper.go

```diff
@@ -1002,42 +1015,121 @@ func (stateMachine *StateMachine) updateGrub(rootfsVolName string, rootfsPartNum
 			relpath:  "/sys",
 			typ:      "sysfs",
 		},
+		{
+			basePath: mountDir,
+			relpath:  "/run",
+			bind:     true,
+		},
 	}
 
-	for _, mp := range mountPoints {
-		mountCmds, umountCmds, err := mp.getMountCmd()
-		if err != nil {
-			return fmt.Errorf("Error preparing mountpoint \"%s\": \"%s\"",
-				mp.relpath,
-				err.Error(),
-			)
-		}
-		updateGrubCmds = append(updateGrubCmds, mountCmds...)
-		teardownCmds = append(umountCmds, teardownCmds...)
+	mountCmds, umountCmds, err := generateMountPointCmds(mountPoints, stateMachine.tempDirs.scratch)
+	if err != nil {
+		return err
 	}
 
+	setupGrubCmds = append(setupGrubCmds, mountCmds...)
+	teardownCmds = append(umountCmds, teardownCmds...)
+
 	teardownCmds = append([]*exec.Cmd{
 		execCommand("udevadm", "settle"),
 	}, teardownCmds...)
 
+	// udev needed to have grub-install properly work
+	grubPackages = append(grubPackages, "udev")
+
+	setupGrubCmds = append(setupGrubCmds,
+		aptInstallCmd(mountDir, grubPackages, false),
+		execCommand("chroot",
+			mountDir,
+			"grub-install",
+			loopUsed,
+			"--boot-directory=/boot",
+			"--efi-directory=/boot/efi",
+			fmt.Sprintf("--target=%s", target),
+			"--uefi-secure-boot",
+			"--no-nvram",
+		),
+	)
+
+	if architecture == arch.AMD64 {
+		setupGrubCmds = append(setupGrubCmds,
+			execCommand("chroot",
+				mountDir,
+				"grub-install",
+				loopUsed,
+				"--target=i386-pc",
+			),
+		)
+	}
+
 	divert, undivert := divertOSProber(mountDir)
 
-	updateGrubCmds = append(updateGrubCmds, divert)
+	setupGrubCmds = append(setupGrubCmds, divert)
 	teardownCmds = append([]*exec.Cmd{undivert}, teardownCmds...)
 
-	// actually run update-grub
-	updateGrubCmds = append(updateGrubCmds,
+	setupGrubCmds = append(setupGrubCmds,
 		execCommand("chroot",
 			mountDir,
 			"update-grub",
 		),
 	)
 
 	// now run all the commands
-	err = helper.RunCmds(updateGrubCmds, stateMachine.commonFlags.Debug)
+	return helper.RunCmds(setupGrubCmds, stateMachine.commonFlags.Debug)
+}
+
+func (stateMachine *StateMachine) confFromArch(architecture string) (string, []string) {
```

Comment:
I feel like this name does not really convey what this method (which doesn't need to be a method?) does

Reply:
Regarding the method aspect: correct! 
Regarding the name: correct too, but so far I failed to find a better one (short and conveying well the job of the method). `targetAndPkgFromArch` could work but is a bit ugly, right? Any other idea?

Reply:
Well I think the name needs to include something about the fact that it is related to grub. grubTargetAndPkgsFromArch seems ok to me. (Although of course maybe it's "Targets" not "Target")